casper-finality-proofs/bin/build_circuits.rs

```diff
@@ -0,0 +1,102 @@
+use casper_finality_proofs::test_engine::wrappers::{
+    compute_shuffled_index::wrapper_mainnet::MAINNET_CIRCUIT as circuit_mainnet,
+    compute_shuffled_index::wrapper_minimal::MINIMAL_CIRCUIT as circuit_minimal,
+    wrapper_weigh_justification_and_finalization::CIRCUIT as circuit_weigh_justification_and_finalization,
+};
+use clap::Parser;
+use once_cell::sync::Lazy;
+use plonky2x::{
+    backend::circuit::{DefaultParameters, GateRegistry, HintRegistry},
+    frontend::builder::CircuitBuilder,
+};
+use std::env::args;
+use strum::{Display, EnumIter, EnumString, IntoEnumIterator};
+#[derive(Debug, Eq, Hash, PartialEq, Copy, Clone, EnumString, Display, EnumIter)]
+#[allow(non_camel_case_types)]
+enum Circuits {
+    compute_shuffled_index_mainnet,
+    compute_shuffled_index_minimal,
+    weigh_justification_and_finalization,
+    all,
+}
+
+impl Circuits {
+    fn from_str(circuit_as_str: &str) -> Option<Circuits> {
+        match circuit_as_str {
+            "compute_shuffled_index_mainnet" => Some(Circuits::compute_shuffled_index_mainnet),
+            "compute_shuffled_index_minimal" => Some(Circuits::compute_shuffled_index_minimal),
+            "weigh_justification_and_finalization" => {
+                Some(Circuits::weigh_justification_and_finalization)
+            }
+            _ => None,
+        }
+    }
+}
+
+enum OneOrAllCircuits {
+    OneCircuit(Box<dyn Fn() -> () + Send + Sync>),
+    AllCircuits(Vec<Box<dyn Fn() -> () + Send + Sync>>),
+}
+
+#[derive(Parser, Debug)]
+struct CommandLineCircuit {
+    /// Enter name of circuit
+    #[clap(value_delimiter = ' ', num_args = 0..)]
+    name: Circuits,
+}
+
+fn build_circuit(circuit: Circuits) -> OneOrAllCircuits {
+    match circuit {
+        Circuits::compute_shuffled_index_mainnet => OneOrAllCircuits::OneCircuit(Box::new(|| {
+            Lazy::force(&circuit_mainnet);
+        })),
+        Circuits::compute_shuffled_index_minimal => OneOrAllCircuits::OneCircuit(Box::new(|| {
+            Lazy::force(&circuit_minimal);
+        })),
+        Circuits::weigh_justification_and_finalization => {
+            OneOrAllCircuits::OneCircuit(Box::new(|| {
+                Lazy::force(&circuit_weigh_justification_and_finalization);
+            }))
+        }
```

Comment:
Return type of build_circuit is never used so it can be removed. Then OneOrAllCircuits becomes obsolete and the Box::new expression can be removed. Simply match the enum value and Lazy::force in the body of the matched value